docs/root/intro/arch_overview/advanced/wasm.rst

```diff
@@ -0,0 +1,70 @@
+.. _arch_overview_wasm:
+
+Wasm
+====
+
+Envoy supports execution of the Wasm modules implemented against the `Proxy-Wasm specification <https://github.com/proxy-wasm/spec>`_.
+Currently, the `binary interface version 0.2.1 <https://github.com/proxy-wasm/spec/tree/main/abi-versions/v0.2.1>`_ is recommended.
+Wasm offers a portable and compact binary executable format that can be compiled once from a variety of languages
+(`C++ <https://github.com/proxy-wasm/proxy-wasm-cpp-sdk>`_, `Rust <https://github.com/proxy-wasm/proxy-wasm-rust-sdk>`_) and
+run anywhere. Please see the :ref:`sandbox example <install_sandboxes_wasm_filter>` for instructions on how to create and deploy
+a Wasm module.
+
+Contexts, plugins, and VMs
+--------------------------
+
+A Wasm context refers to the interface implementation between the host (Envoy) and the guest (Wasm bytecode). For an
+HTTP filter, there are two types of contexts: a *root context* (context with ID 0) represents the configuration-only
+context, and a *per-request context* that is created once per each request.
+
+A Wasm VM refers to the execution instance of a Wasm bytecode. Multiple VMs can be created for the same bytecode when
```

Comment:
Saying that a Wasm VM is the execution instance of Wasm bytecode could be confusing when applied to multiple Wasm plugins. How about rewording to:

```
A Wasm VM refers to an instance of a Wasm runtime that executes Wasm
plugins. When configuring Wasm plugins, the vm_id field can be used to control
whether plugins run in the same Wasm VM as each other, or separate ones. VM
sharing optimizes the runtime overhead and reduces the total binary size of the
bytecode since the extensions can share the code.

```